wolfcrypt/src/random.c

```diff
@@ -2140,6 +2141,41 @@ int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
 
     #define USE_TEST_GENSEED
 
+#elif defined(WOLFSSL_ZEPHYR)
+
+        #include <entropy.h>
+    #ifndef _POSIX_C_SOURCE
+        #include <posix/time.h>
+    #else
+        #include <sys/time.h>
+    #endif
+
+        int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
+        {
+            int ret = 0;
+            struct device* dev;
+
+            dev = device_get_binding(CONFIG_ENTROPY_NAME);
+            if (dev != NULL) {
+                if (entropy_get_entropy(dev, output, sz) != 0)
+                    ret = READ_RAN_E;
+            }
```

Comment:
Does the dev device need to be closed?

Reply:
Nope. Only binding to it - getting a handle to the device.